src/nvim/eval.c

```diff
@@ -8891,6 +8913,7 @@ static void filter_map(typval_T *argvars, typval_T *rettv, int map)
       }
       hash_unlock(ht);
     } else {
+      // argvars[0].v_type == VAR_LIST
```

Comment:
Why is this comment instead of an assert or `} else if { /* code for list type */ } else { /* throw error */ }`?

Reply:
Vim almost never uses `assert()` AFAIR.